pkg/repository/ensurer.go

```diff
@@ -130,14 +136,14 @@ func (r *RepositoryEnsurer) EnsureRepo(ctx context.Context, namespace, volumeNam
 
 	log.Debug("Acquired lock")
 
-	selector := labels.SelectorFromSet(repoLabels(volumeNamespace, backupLocation))
+	selector := labels.SelectorFromSet(repoLabels(volumeNamespace, backupLocation, repositoryType))
```

Comment:
For the existing `BackupRepository` for the older velero version, If the user didn't reinstall the velero 1.10, we cannot list `BackupRepository` by the new selector. because the new label is not added to the older `BackupRepository`. 

So for users who upgrade from an older version to 1.10, do they need to delete all crds and reuninstall velero?

Reply:
The legacy repos are not called `BackupRepository` but `ResticRepository`, therefore, we cannot list the legacy repos anyway.
And after the upgrade, the legacy repos named as `ResticRepository` will not be used anymore, for the old PVB/PVR, a new `BackupRepository` will be created; and the controller doesn't handle them in its reconciler either.

Therefore it is safe to leave it there and we need users to delete them manually.